be/src/olap/rowset/segment_v2/index_page.h

```diff
@@ -0,0 +1,128 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include "common/status.h"
+#include "gen_cpp/segment_v2.pb.h"
+#include "gutil/macros.h"
+#include "olap/rowset/segment_v2/page_pointer.h"
+#include "util/faststring.h"
+#include "util/slice.h"
+
+namespace doris {
+namespace segment_v2 {
+
+class IndexPageIterator; // forward decl.
+
+class IndexPageBuilder {
```

Comment:
Should add more comments for IndexPage format.

And I have another question, which is why not use already exist PageBuilder?

Reply:
> Should add more comments for IndexPage format.
Sure :-)

> why not use already exist PageBuilder?
Good question. PageBuilder is designed for data page but IndexPageBuilder is designed for index page. Index page is fundamentally different from data page because it is a key-value structure. Index page provides interface to search value by key, which can't be done for data page because it has no concept of key and value.